src/main/java/seedu/realodex/logic/parser/AddCommandParser.java

```diff
@@ -37,106 +41,157 @@ public class AddCommandParser implements Parser<AddCommand> {
+     * @return The parsed AddCommand object.
+     * @throws ParseException If the user input does not conform to the expected format.
      */
     public AddCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_INCOME, PREFIX_EMAIL, PREFIX_ADDRESS,
-                        PREFIX_FAMILY, PREFIX_TAG, PREFIX_HOUSINGTYPE, PREFIX_REMARK, PREFIX_BIRTHDAY);
-
-        Prefix[] listOfCompulsoryPrefixTags = argMultimap.returnListOfCompulsoryTags(PREFIX_NAME, PREFIX_ADDRESS,
-                                                                                     PREFIX_INCOME, PREFIX_PHONE,
-                                                                                     PREFIX_FAMILY, PREFIX_EMAIL,
-                                                                                     PREFIX_TAG, PREFIX_HOUSINGTYPE);
-        if (!arePrefixesPresent(argMultimap,
-                                listOfCompulsoryPrefixTags)) {
-            String exceptionMessageForMissingPrefixes =
-                    argMultimap.returnMessageOfMissingPrefixes(listOfCompulsoryPrefixTags) + "\n";
-            throw new ParseException(Messages.getErrorMessageForMissingPrefixes(exceptionMessageForMissingPrefixes)
+        ArgumentMultimap argMultimap = tokenizeArguments(args);
+
+        validateCompulsoryPrefixes(argMultimap);
+
+        validateNoDuplicatePrefixes(argMultimap);
+
+        Person person = buildPerson(argMultimap);
+
+        return new AddCommand(person);
+    }
+
+    /**
+     * Tokenizes the input arguments.
+     *
+     * @param args The arguments provided by the user.
+     * @return An ArgumentMultimap containing the parsed arguments.
+     */
+    private ArgumentMultimap tokenizeArguments(String args) {
+        return ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_INCOME, PREFIX_EMAIL, PREFIX_ADDRESS,
+                                          PREFIX_FAMILY, PREFIX_TAG, PREFIX_HOUSINGTYPE,
+                                          PREFIX_REMARK, PREFIX_BIRTHDAY);
+    }
+
+    /**
+     * Validates the presence of compulsory prefixes.
+     *
+     * @param argMultimap An ArgumentMultimap containing the parsed arguments.
+     * @throws ParseException If any compulsory prefix is missing.
+     */
+    private void validateCompulsoryPrefixes(ArgumentMultimap argMultimap) throws ParseException {
+        Prefix[] compulsoryPrefixes = {PREFIX_NAME, PREFIX_ADDRESS, PREFIX_INCOME, PREFIX_PHONE,
+            PREFIX_FAMILY, PREFIX_EMAIL,
+            PREFIX_TAG, PREFIX_HOUSINGTYPE};
+        if (!arePrefixesPresent(argMultimap, compulsoryPrefixes)) {
+            String missingPrefixesMessage = argMultimap.returnMessageOfMissingPrefixes(compulsoryPrefixes) + "\n";
+            throw new ParseException(Messages.getErrorMessageForMissingPrefixes(missingPrefixesMessage)
```

Comment:
Could you make the parameters all on the same line?
[BIRTHDAY] should be on the same line as the rest
<img width="1461" alt="image" src="https://github.com/AY2324S2-CS2103T-W10-1/tp/assets/59839013/01b96e3e-4487-43bd-a9cb-dfdee6c01199">

Reply:
Perhaps Parameters should start in a new line too

Reply:
![image](https://github.com/AY2324S2-CS2103T-W10-1/tp/assets/39399745/7c514a9a-8d9b-4f3a-81b7-dac186989a10)

Thanks